src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -81,6 +82,7 @@ public enum UnreferencedTypesHandlingOption
             IReadOnlyList<string> modelsToTreatEmptyStringAsNull,
             IReadOnlyList<string> additionalIntrinsicTypesToTreatEmptyStringAsNull,
             bool shouldTreatBase64AsBinaryData,
+            IReadOnlyList<string> methodsToKeepClientDefaultValue,
```

Comment:
I thought this should be a global setting that applies to all methods, but now seems this is a per method setting.

What is the main thinking behind this?

Reply:
We only suppress the methods which has breaking change now. In the future, if any new operation has client default value we still need to eliminate the default value. So we need suppress on method-level, not the service-level which will apply for all methods(current and new coming).